server/aws-lsp-codewhisperer/src/language-server/agenticChat/tools/qCodeAnalysis/displayFindings.ts

```
@@ -0,0 +1,150 @@
+    private readonly telemetry: Features['telemetry']
+    private readonly workspace: Features['workspace']
+    private cancellationToken?: CancellationToken
+    private writableStream?: WritableStream
+
+    constructor(features: Pick<Features, 'logging' | 'telemetry' | 'workspace'> & Partial<Features>) {
+        this.logging = features.logging
+        this.telemetry = features.telemetry
+        this.workspace = features.workspace
+    }
+
+    static readonly toolName = DISPLAY_FINDINGS_TOOL_NAME
+
+    static readonly toolDescription = DISPLAY_FINDINGS_TOOL_DESCRIPTION
+
+    static readonly inputSchema = DISPLAY_FINDINGS_INPUT_SCHEMA
+
+    /**
+     * Main execution method for the displayFindings tool
+     * @param input User input parameters for display findings
+     * @param context Execution context containing clients and tokens
+     * @returns Output containing code review results or error message
+     */
+    public async execute(input: any, context: any): Promise<InvokeOutput> {
+        let chatStreamWriter: WritableStreamDefaultWriter<any> | undefined
+
+        try {
+            this.logging.info(`Executing ${DISPLAY_FINDINGS_TOOL_NAME}: ${JSON.stringify(input)}`)
+
+            // 1. Validate input
+            const setup = await this.validateInputAndSetup(input, context)
+            this.checkCancellation()
+
+            // 2. group the findings into AggregatedCodeScanIssue
+            const mappedFindings = setup.map(finding => this.mapToCodeReviewFinding(finding))
+            const aggregatedFindings = this.aggregateFindingsByFile(mappedFindings)
+
+            return {
+                output: {
+                    kind: 'json',
+                    success: true,
+                    content: aggregatedFindings,
+                },
+            }
+        } catch (error: any) {
+            if (error instanceof CancellationError) {
+                throw error
+            }
+            return {
```

Dont return it, the agent doesn't recognize that the tool failed. We need to throw the exception. Replace this return inside catch with `throw new Error(error.message)`

I fixed it for codeReview tool

---

Yes